
В качестве набора данных будем использовать рейтинг популярных пабликов с мемами про котиков. Информация собрана в таблице ниже:
```
Место в рейтинге	Число постов	Количество лайков	Количество репостов

1	                280	            892	                621

2	                254	            742	                763

3	                352	            820	                757

4	                189	            730	                614

5	                209	            758	                608

6	                312	            652	                545

7	                199	            680	                553

8	                224	            592	                485

9	                293	            603	                466

10	                187	            507	                423

```

In [14]:
import numpy as np
rating = np.array([[1,280,892,621],
                    [2,254,742,763],
                    [3,352,820,757],
                    [4,189,730,614],
                    [5,209,758,608],
                    [6,312,652,545],
                    [7,199,680,553],
                    [8,224,592,485],
                    [9,293,603,466],
                    [10,187,507,423]])
"""
Вычислить среднее арифметическое для количества репостов, то есть сумму репостов во всех пабликах разделить на число пабликов в рейтинге.
"""

mean = np.mean(rating[:,-1])
mean

np.float64(583.5)

In [15]:
#Медиана если не имеет большого отклонения от среднего, значит нет больших выбросов
median = np.median(rating[:,-1])
median

np.float64(580.5)

array([[1.        , 0.38492604],
       [0.38492604, 1.        ]])

In [17]:
mean = np.mean(rating[:,1])
mean

np.float64(249.9)

In [18]:
#Вычисление стандартного отклонения
std = np.std(rating[:,1])
std

np.float64(54.46182148992081)

Корреляция (corrcoef) — статистическая взаимосвязь между переменными.
Корреляция показывает степень связи между двумя явлениями, меняются переменные совместно или нет. 
Значение коэффициента корреляции показывает, насколько сильно связаны между собой переменные и всегда принимает значение от 0 до 1 (по модулю). Коэффициент может быть положительным или отрицательным.


In [19]:
corr = np.corrcoef(rating[:,1],rating[:,2])
corr

array([[1.        , 0.38492604],
       [0.38492604, 1.        ]])

#Пояснение вывода
     0         1
0  1.0     0.384926  (X с X и X с Y)
1  0.384926  1.0     (Y с X и Y с Y)

Это таблица корреляции, которая показывает, как связаны между собой разные переменные. В вашем случае у вас две переменные:

Число постов (обозначим как X)
Количество лайков (обозначим как Y)
Таблица выглядит следующим образом:

Пояснение:
Элементы на главной диагонали (1.0):

1.0 в первой ячейке (0, 0) обозначает корреляцию переменной X с самой собой (число постов с числом постов), что всегда будет равно 1.
То же относится ко второй ячейке (1, 1) для переменной Y (количество лайков с самим собой).
Элементы вне главной диагонали (0.384926):

Значение 0.384926 в ячейке (0, 1) — это коэффициент корреляции между числом постов (X) и количеством лайков (Y).
Это значение указывает на то, что есть умеренная положительная корреляция между этими переменными. То есть, когда увеличивается число постов, количество лайков тоже, в общем, увеличивается, но не идеально.
